--- app/src/main/java/org/ole/planet/myplanet/ui/courses/CoursesViewModel.kt
+++ app/src/main/java/org/ole/planet/myplanet/ui/courses/CoursesViewModel.kt
@@ -95,11 +95,13 @@

     fun filterCourses(isMyCourseLib: Boolean, userId: String?, searchText: String, selectedGrade: String, selectedSubject: String, tagNames: List<String>) {
         viewModelScope.launch {
-            val filteredCourses = coursesRepository.filterCourses(searchText, selectedGrade, selectedSubject, tagNames)
-            val myCourses = filteredCourses.filter { it.userId?.contains(userId) == true }
-            val tagsMap = _coursesState.value.tagsMap
-            processCourses(isMyCourseLib, userId, filteredCourses, myCourses, _coursesState.value.map, _coursesState.value.progressMap, tagsMap)
+            withContext(dispatcherProvider.io) {
+                val filteredCourses = coursesRepository.filterCourses(searchText, selectedGrade, selectedSubject, tagNames)
+                val myCourses = filteredCourses.filter { it.userId?.contains(userId) == true }
+                val tagsMap = _coursesState.value.tagsMap
+                processCourses(isMyCourseLib, userId, filteredCourses, myCourses, _coursesState.value.map, _coursesState.value.progressMap, tagsMap)
+            }
         }
     }
