--- app/src/main/java/org/ole/planet/myplanet/ui/notifications/NotificationsFragment.kt
+++ app/src/main/java/org/ole/planet/myplanet/ui/notifications/NotificationsFragment.kt
@@ -75,9 +75,14 @@
         val spinnerAdapter = ArrayAdapter(requireContext(), R.layout.spinner_item, optionsList)
         spinnerAdapter.setDropDownViewResource(R.layout.spinner_item)
         binding.status.adapter = spinnerAdapter
+        var isInitialSpinnerSelection = true
         binding.status.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
             override fun onItemSelected(parent: AdapterView<*>, view: View?, position: Int, id: Long) {
+                if (isInitialSpinnerSelection) {
+                    isInitialSpinnerSelection = false
+                    return
+                }
                 currentFilter = parent.getItemAtPosition(position).toString().lowercase()
                 viewModel.loadNotifications(userId, currentFilter, isAdmin)
             }
