--- app/src/main/java/org/ole/planet/myplanet/ui/health/HealthExaminationAdapter.kt
+++ app/src/main/java/org/ole/planet/myplanet/ui/health/HealthExaminationAdapter.kt
@@ -141,13 +141,8 @@
         try {
             val conditionsMap = JsonUtils.gson.fromJson(realmExamination.conditions, JsonObject::class.java)
             if (conditionsMap != null) {
-                val keys = conditionsMap.keySet()
-                val conditionsBuilder = StringBuilder()
-                for (key in keys) {
-                    if (conditionsMap[key].asBoolean) {
-                        conditionsBuilder.append("$key, ")
-                    }
-                }
-                conditionsText = conditionsBuilder.toString()
+                conditionsText = conditionsMap.keySet()
+                    .filter { conditionsMap[it].asBoolean }
+                    .joinToString(", ")
             }
         } catch (e: Exception) {
