--- ./app/src/main/java/org/ole/planet/myplanet/repository/ProgressRepositoryImpl.kt
+++ ./app/src/main/java/org/ole/planet/myplanet/repository/ProgressRepositoryImpl.kt
@@ -23,7 +23,6 @@
 class ProgressRepositoryImpl @Inject constructor(
     databaseService: DatabaseService,
     @RealmDispatcher realmDispatcher: CoroutineDispatcher,
-    private val dispatcherProvider: DispatcherProvider,
     private val coursesRepositoryLazy: dagger.Lazy<CoursesRepository>,
     private val activitiesRepositoryLazy: dagger.Lazy<ActivitiesRepository>
 ) : RealmRepository(databaseService, realmDispatcher), ProgressRepository {
@@ -224,8 +223,8 @@
         }
     }

-    override suspend fun hasUserCompletedSync(userId: String): Boolean = withContext(dispatcherProvider.io) {
-        activitiesRepositoryLazy.get().hasUserCompletedSync(userId)
+    override suspend fun hasUserCompletedSync(userId: String): Boolean {
+        return activitiesRepositoryLazy.get().hasUserCompletedSync(userId)
     }

     private fun insertCourseProgress(
