--- ./app/src/main/java/org/ole/planet/myplanet/ui/exam/UserInformationFragment.kt
+++ ./app/src/main/java/org/ole/planet/myplanet/ui/exam/UserInformationFragment.kt
@@ -48,12 +48,8 @@
     var userModel: RealmUser? = null
     var shouldHideElements: Boolean? = null
     @Inject
-    lateinit var uploadManager: UploadManager
-    @Inject
-    lateinit var submissionUploadExecutor: SubmissionUploadExecutor
-    @Inject
-    lateinit var serverUrlMapper: ServerUrlMapper
-    @Inject
     lateinit var sharedPrefManager: SharedPrefManager
+    @Inject
+    lateinit var submissionsSyncCoordinator: SubmissionsSyncCoordinator
     private var syncStartTime: Long = 0L
