--- ./app/src/main/java/org/ole/planet/myplanet/ui/sync/SyncActivity.kt
+++ ./app/src/main/java/org/ole/planet/myplanet/ui/sync/SyncActivity.kt
@@ -194,7 +194,7 @@
                         val canReachServer = MainApplication.isServerReachable(serverUrl)
                         if (canReachServer) {
                             withContext(dispatcherProvider.main) {
-                                startUpload("login")
+                                syncWorkRepository.uploadLoginData()
                             }
                             transactionSyncManager.syncDb("login_activities")
                         }
